src/testrun.rs

```diff
@@ -111,3 +128,82 @@ impl Testrun {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[pyclass(eq, eq_int)]
+pub enum Framework {
+    Pytest,
+    Vitest,
+    Jest,
+    PHPUnit,
+}
+
+#[pymethods]
+impl Framework {
+    #[new]
+    #[pyo3(signature = (value=None))]
+    fn new(value: Option<&str>) -> Self {
+        match value {
+            Some("pytest") => Framework::Pytest,
+            Some("vitest") => Framework::Vitest,
+            Some("jest") => Framework::Jest,
+            Some("phpunit") => Framework::PHPUnit,
+            Some(_) => panic!("this should not occur"), // TODO error message here
+            None => panic!("this should not occur"),
+        }
+    }
```

Comment:
Instead of making this a constructor with `fn new(Option<&str>) -> Self`, make this a class method, like `fn from_str(s: &str) -> Option<Self>`, then you can properly type this, and do not have to resort to panics.